File: frontend/src/api/lists.js
/**
 * lists.js - List API Functions
 * 
 * Contains all API calls related to lists:
 * - Creating, updating, deleting lists
 * - Reordering lists
 */

import api from './axios';

/**
 * Create a new list in a board
 * @param {string} title - List title
 * @param {number} boardId - The board ID
 * @param {number} position - List position (order)
 */
export const createList = (title, boardId, position = 0) => 
  api.post('/lists/', { title, board_id: boardId, position });

/**
 * Update a list (title or position)
 * @param {number} listId - The list ID
 * @param {object} data - Fields to update
 */
export const updateList = (listId, data) => 
  api.patch(`/lists/${listId}`, data);

/**
 * Delete a list
 * @param {number} listId - The list ID to delete
 */
export const deleteList = (listId) => 
  api.delete(`/lists/${listId}`);

/**
 * Move/reorder a list
 * @param {number} listId - The list ID
 * @param {number} position - New position
 */
export const moveList = (listId, position) => 
  api.patch(`/lists/${listId}`, { position });

File: backend/app/api/lists.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.schemas.list import ListCreate, ListOut, ListUpdate, ListMove
from app.crud.list import create_list, get_list, update_list, delete_list, reorder_lists
from app.database import get_db

router = APIRouter(prefix="/lists", tags=["lists"])

@router.post("/", response_model=ListOut)
def create(data: ListCreate, db: Session = Depends(get_db)):
    """Create a new list in a board"""
    return create_list(db, data)

@router.get("/{list_id}", response_model=ListOut)
def read(list_id: int, db: Session = Depends(get_db)):
    """Get a specific list with its cards"""
    db_list = get_list(db, list_id)
    if not db_list:
        raise HTTPException(status_code=404, detail="List not found")
    return db_list

@router.patch("/{list_id}", response_model=ListOut)
def update(list_id: int, data: ListUpdate, db: Session = Depends(get_db)):
    """Update a list (title or position)"""
    db_list = update_list(db, list_id, data)
    if not db_list:
        raise HTTPException(status_code=404, detail="List not found")
    return db_list

@router.delete("/{list_id}")
def delete(list_id: int, db: Session = Depends(get_db)):
    """Delete a list and all its cards"""
    if not delete_list(db, list_id):
        raise HTTPException(status_code=404, detail="List not found")
    return {"message": "List deleted successfully"}

@router.patch("/{list_id}/move", response_model=ListOut)
def move(list_id: int, data: ListMove, db: Session = Depends(get_db)):
    """Reorder a list (drag and drop)"""
    db_list = get_list(db, list_id)
    if not db_list:
        raise HTTPException(status_code=404, detail="List not found")
    
    result = reorder_lists(db, db_list.board_id, list_id, data.position)
    if not result:
        raise HTTPException(status_code=400, detail="Failed to move list")
    return result
